app/src/androidTest/java/com/github/se/polyfit/data/api/SpoonacularApiCallerTest.kt

```diff
@@ -282,11 +294,114 @@ class SpoonacularApiCallerTest {
     assertEquals(response.status, APIResponse.FAILURE)
   }
 
+  @Test
+  fun recipeStepFromId() {
+    val response = spoonacularApiCaller.getRecipeSteps(1)
+
+    assertEquals("cheese", response.name)
+    assertEquals(2, response.steps.size)
+    assertEquals(1, response.steps[0].number)
+  }
+
+  @Test
+  fun recipeStepFromIdFailure() {
+    mockWebServer.dispatcher = faultyDispatcher
+
+    val response = spoonacularApiCaller.getRecipeSteps(1)
+
+    assertEquals(response, RecipeInstruction.faillure())
+  }
+
+  @Test
+  fun getCompleteRecipesFromIngredients() {
+    val recipe = spoonacularApiCaller.getCompleteRecipesFromIngredients(listOf("apple", "banana"))
+
+    assert(recipe.isNotEmpty())
+    assert(recipe.first().recipeInformation.ingredients.isNotEmpty())
+    assert(recipe.first().recipeInformation.instructions.isNotEmpty())
+  }
+
+  @Test
+  fun getCompleteRecipesFromIngredientsFaillure() {
+    mockWebServer.dispatcher = faultyDispatcher
+
+    val response = spoonacularApiCaller.getCompleteRecipesFromIngredients(listOf("apple", "banana"))
+
+    assert(response.first().recipeInformation.ingredients.isEmpty())
+    assert(response.first().recipeInformation.instructions.isEmpty())
+  }
+
   @After
   fun tearDown() {
     mockWebServer.shutdown()
   }
 
+  private val recipeStepsMock =
```

Comment:
id maybe suggest to create a folder in the test files called mockData and include this and large json test data there to keep the test file more logic focused

Reply:
Great suggestion !